core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -978,9 +983,10 @@ class ReplicaManager(val config: KafkaConfig,
 
       val adjustedMaxBytes = math.min(fetchInfo.maxBytes, limitBytes)
       try {
-        trace(s"Fetching log segment for partition $tp, offset $offset, partition fetch size $partitionFetchSize, " +
-          s"remaining response limit $limitBytes" +
-          (if (minOneMessage) s", ignoring response/partition size limits" else ""))
+        if (isTraceEnabled)
```

Comment:
Looking up the trace level above means we only need to get the effective level once even though read() may be called once for each partition.